localstack/services/apigateway/helpers.py

```diff
@@ -664,7 +664,10 @@ def add_path(path, parts, parent_id=""):
             parent_id=parent_id,
         )
 
-        for method, method_schema in resolved_schema["paths"].get(path, {}).items():
+        paths_dict = resolved_schema["paths"]
+        relative_path = path.removeprefix(base_path)
```

Comment:
The all PR seems to be related with this `base_path` removal, but unless this is "swagger" specific I don't understand how this is working for OpenAPI, since this [line](https://github.com/localstack/localstack/blob/d45401861045ac42a4e2c519b163f59999cbac63/localstack/services/apigateway/helpers.py#L733) adds specifically the base path in case it's in the "prepend" mode.

Reply:
Good catch @calvernaz , my bad - renamed the PR title (it is in fact for Swagger, at least for this use case).

Do you have an example where this could be causing issues (e.g., with `basePath` value as `ignore` or `split`)..? My assumption was that the paths in `paths` are always referring to the relative paths (independent of `basePath`), but we'll probably need to run a few more checks against real AWS. 

FWIW, the example is a valid SAM template, and the corresponding CFn resource that gets created looks like this at runtime:

```
{'Type': 'AWS::ApiGateway::RestApi', 'LogicalResourceId': 'Api', 'Properties': {'Body': {'info': {'version': '1.0', 'title': 'Public API'}, 'paths': {'/test': {'post': {'x-amazon-apigateway-integration': {'httpMethod': 'POST', 'type': 'http_proxy', 'uri': 'IntegrationUri'}, 'responses': {}}}}, 'schemes': ['https'], 'basePath': '/prod', 'x-amazon-apigateway-binary-media-types': ['*/*'], 'swagger': 2.0}, 'Name': 'stack-20bba16f-Api-498e561d'}, '_state_': {}}
```

Reply:
This was a great pointer @calvernaz that led to a bit more investigation, as discussed. Updated the PR - the new test is now `@pytest.mark.aws_validated`, and one of the existing tests has been parameterized with the different basepath types ("ignore", "prepend", "split")

Reply:
The tests look great and thanks for adding the other modes :100: 

My only comment then is if the `removeprefix` can be done outside the `add_path` method. Basically the lookup is done using correct path.

Reply:
Thanks, good point @calvernaz  - updated the PR to make the logic and variable names of relative/absolute/base paths a bit more concrete. 👍

Reply:
🥇